### PR TITLE
feat: unsplit wrapException with improved types & runtime

### PR DESCRIPTION
**Pull request by sergioflores-j**
Proposal: have only one exported member for wrapException.
Pros: only one function to care about for consumers
Downsides: as there are "unlimited crazy" ways for making promises run, our type guard might not be very safe and accurate. Which might cause issues during runtime (e.g. unhandled promises, etc)

### CODE REVIEW

**Comment by r5402** [comment]
> Which might cause issues during runtime (e.g. unhandled promises, etc)

Can you elaborate? Can you give any examples?

**Comment by sergioflores-j** [comment]
> > Which might cause issues during runtime (e.g. unhandled promises, etc)
> 
> Can you elaborate? Can you give any examples?

I can't give any examples as I don't really know any other edge cases. I tried covering the most common scenarios in the test cases.
By reading multiple forums, I noticed people tend to say that is not very safe to assert promises most of the times, as there are more implementations than the native one that can be used and so on.

From promise's docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
> [Promise.resolve()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve)
[...]
Generally, if you don't know if a value is a promise or not, [Promise.resolve(value)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) it instead and work with the return value as a promise.
> 

I made it as safe as possible, but executing the "non async" function to check if it's return is "PromiseLike" or not, but who knows what else is possible haha /shrug

File: src/wrapException/index.ts [hunk 1]
```diff
@@ -68,15 +78,69 @@ export function wrapException<Args extends unknown[], Response>(
...
+          typeof fnExec === 'function' ? fnExec(...args) : fnExec,
+        )
+          .then((r) => [undefined, r] as const)
+          // Catches rejects from Promises & throws from AsyncFunctions - avoids "Unhandled promise rejection"
+          .catch((error: unknown) => [error, undefined] as const);
+
+        return result;
+      } catch (error: unknown) {
+        // Catches errors from the above execution
+        return [error, undefined] as const;
```
Comment (sergioflores-j) [inline src/wrapException/index.ts:121]: Nothing actually goes here, I just did it to be extra safe... couldn't think of any possible scenario anymore...

I recall something like this happened in the past with me, especially when mapping lists, etc.

In cases like if you had a `new Promise` that was neither rejected nor resolved, but actually an error was being thrown. (think of it like a method that does "axios.post" but has validations that throw something)

**Review by r5402 — APPROVED** [review]
It looks great! Good job.

File: src/wrapException/index.ts (hunk 1 above)
Comment (r5402) [reply src/wrapException/index.ts:121]: We'll probably hit some more edge cases as soon as we get some folks to use it, maybe.

I will do some more research on top of it also.
